### ISSUE TITLE
BlueDot.btcomm 'threading' data_received calls

### ISSUE THREAD

**Issue by martinohanlon** [martinohanlon/BlueDot#80]
BlueDot.btcomm doesnt spawn threads for running the 'data_received' function. This results in potential backlog issues whereby BlueDot is still processing calls when new messages are received.

**Comment by martinohanlon** [comment]
Will need to modify:

```
_process_commands
_pressed
_double_pressed
_released
_moved
_process_callback
```

to make `_position` thread safe, otherwise the position which is passed back may not be the actual position.

**Comment by martinohanlon** [comment]
A change to allow call_backs from BlueDot to be threaded is included in v1.1.0
